be1-go/hub/organizer.go

```diff
@@ -484,3 +502,161 @@ func (c *laoChannel) processRollCallObject(data message.Data) error {
 
 	return nil
 }
+func (c *laoChannel) processElectionObject(msg message.Message) error {
+	action := message.ElectionAction(msg.Data.GetAction())
+
+	if action != message.ElectionSetupAction {
+		return &message.Error{
+			Code:        -1,
+			Description: fmt.Sprintf("invalid action: %s", action),
+		}
+	}
+	err := c.createElection(msg)
+	if err != nil {
+		return xerrors.Errorf("failed to setup the election", err)
+	}
+
+	return nil
+}
+
+func (c *laoChannel) createElection(msg message.Message) error {
+	o := c.hub
+
+	o.Lock()
+	defer o.Unlock()
+
+	// Check the data
+	data, ok := msg.Data.(*message.ElectionSetupData)
+
+	if !ok {
+		return &message.Error{
+			Code:        -4,
+			Description: "failed to cast data to SetupElectionData",
+		}
+	}
+
+	// Check if the Lao ID of the message corresponds to the channel ID
+	encodedLaoID := base64.StdEncoding.EncodeToString(data.LaoID)
+	channelID := c.channelID[6:]
+	if channelID != encodedLaoID {
+		return &message.Error{
+			Code:        -4,
+			Description: fmt.Sprintf("Lao ID of the message (Lao: %s) is different from the channelID (channel: %s)", encodedLaoID, channelID),
+		}
+	}
+
+	// Compute the new election channel id
+	encodedElectionID := base64.StdEncoding.EncodeToString(data.ID)
+	encodedID := encodedLaoID + "/" + encodedElectionID
+
+	// Create the new election channel
+	electionCh := electionChannel{
+		createBaseChannel(o, "/root/"+encodedID),
+		data.Questions[0].VotingMethod,//TODO : check if this is what was meanth by method ot is it rather pluralityy and stuff??,
+		//Also can it happen that there is a valid setup without questions?
```

Comment:
According to the [protocol](https://github.com/dedis/student_21_pop/blob/ef74df3bbf0f8efb5389ecb80f5080b45a5016cd/protocol/query/method/message/data/dataElectionSetup.json#L81) there is at least one question per election.